crates/database/src/lib.rs

```diff
@@ -52,6 +52,18 @@ where
         Ok(entities)
     }
 
+    /// Counts all entities by filter.
+    async fn count_by(&self, filter: BsonDocument) -> Result<usize, RepositoryError> {
+        let collection = self.get_collection();
+        // Lock the Mutex and get the Collection
+        let collection = collection.lock().await;
+        Ok(collection
+            .count_documents(filter, None)
+            .await?
+            .try_into()
+            .map_err(|_| RepositoryError::Generic("count overflow".to_owned()))?)
+    }
```

Comment:
It is actually just a precaution. i64 is actually a very big number. Not sure that the database will be allowed to store that number of messages for a single recipient. So it will probably never happen.

Reply:
Okay, it makes sense to not bother further.